src/clr-angular/forms/checkbox/checkbox.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Directive, ViewContainerRef} from "@angular/core";
+
+import {WrappedFormControl} from "../common/wrapped-form-control";
+
+import {ClrCheckboxContainer} from "./checkbox-container";
+
+@Directive({selector: "[clrCheckbox]"})
+export class ClrCheckboxNext extends WrappedFormControl<ClrCheckboxContainer> {
+    // Once again, several more elegant solutions were foiled by severity 3+ bugs on Angular that have been opened
+    // for 6 months to a year. So that's how we do it. Inheritance and ridiculous constructors. :-(
```

Comment:
Please point to this ticket in the comment, otherwise we might lose track of it.

Reply:
I have subscriptions on the issues, but as mentioned before it's impossible to find tickets you're subscribed to... Anyway, once the newest compiler hits, maybe we'll want to do a pass on all of our abstract classes and inheritance patterns. But it's not going to be a priority.